tests/ui/job-view/Filtering_test.jsx

```diff
@@ -0,0 +1,275 @@
+        getAllByText('View Tests'),
+      );
+      expect(filteredPushes).toHaveLength(1);
+
+      setUrlParam('author', null);
+      await waitForElement(() => getAllByText('jarilvalenciano@gmail.com'));
+      const unFilteredPushes = await waitForElement(() =>
+        getAllByText('View Tests'),
+      );
+      expect(unFilteredPushes).toHaveLength(10);
+    });
+  });
+
+  describe('by failure result', () => {
+    beforeEach(() => {
+      // polling..  Return nothing...
+      fetchMock.get(
+        `begin:${getProjectUrl(
+          '/push/?full=true&count=11&push_timestamp__lte=',
+          'autoland',
+        )}`,
+        {
+          results: [],
+        },
+      );
+    });
+
+    test('should have 20 failures', async () => {
+      const { getAllByText, getByTitle, findAllByText } = render(<App />);
+      await waitForElement(() => findAllByText('B'));
+      const unclassifiedOnlyButton = getByTitle(
+        'Loaded failures / toggle filtering for unclassified failures',
+      );
+      fireEvent.click(unclassifiedOnlyButton);
+
+      await waitForElementToBeRemoved(() => getAllByText('yaml'));
+
+      // The api returns the same joblist for each push.
+      // 10 pushes with 2 failures each, but only 1 unclassified.
+      expect(jobCount()).toBe(10);
+
+      // undo the filtering and make sure we see all the jobs again
+      fireEvent.click(unclassifiedOnlyButton);
+      await waitForElement(() => getAllByText('yaml'));
+      expect(jobCount()).toBe(40);
+    });
+  });
+
+  describe('by keywords', () => {
+    beforeAll(() => {
+      fetchMock.get(
+        getProjectUrl('/jobs/259537372/', 'autoland'),
+        Object.values(jobMap)[0],
+      );
+      fetchMock.get(
+        getProjectUrl('/job-log-url/?job_id=259537372', 'autoland'),
+        [],
+      );
+      fetchMock.get(
+        getProjectUrl('/performance/data/?job_id=259537372', 'autoland'),
+        [],
+      );
+      fetchMock.get(getApiUrl('/jobdetail/?job_id=259537372'), { results: [] });
+      fetchMock.get(getProjectUrl('/note/?job_id=259537372', 'autoland'), []);
+      fetchMock.get(
+        getProjectUrl('/bug-job-map/?job_id=259537372', 'autoland'),
+        [],
+      );
+      fetchMock.get(
+        getProjectUrl('/jobs/259537372/bug_suggestions/', 'autoland'),
+        [],
+      );
+      fetchMock.get(
+        getProjectUrl('/jobs/259537372/text_log_steps/', 'autoland'),
+        [],
+      );
+    });
+
+    const setFilterText = (filterField, text) => {
+      fireEvent.click(filterField);
+      fireEvent.change(filterField, { target: { value: text } });
+      fireEvent.keyDown(filterField, { key: 'Enter', code: 13 });
+    };
+
+    test('string "yaml" should have 10 jobs', async () => {
+      const { getAllByText, findAllByText } = render(<App />);
+      await waitForElement(() => findAllByText('B'));
+      const filterField = document.querySelector('#quick-filter');
```

Comment:
You could move this within `setFilterText` and drop the first agument `filterField`.
This is just a suggestion; whatever you prefer.

Reply:
I actually reuse that ``const`` on line 202 as well, so it's worth keeping.

Reply:
I'm curious you use `document.querySelector` here instead of one of the react-testing-library methods?

Reply:
I could be wrong, but I don't think the react-testing-library querying supports searching by ``id``.  I know you can do ``getByTestId``, but then I'd have to add a test id to each element.  Am I missing it?  :)

Reply:
You might be able to use `toHaveAttribute` or if you changed it to a class `toHaveClass` (if it's used in more than one place maybe it'd make sense to change to a class name anyways?): https://github.com/testing-library/jest-dom#tohaveattribute